### PR TITLE
Add: pyAMReX

### PR DESCRIPTION
**Pull request by ax3l**
Add the new pybind11 project `pyAMReX` as `py-amrex` package.

Depends on
- [x] #31913

### CODE REVIEW

**Review by adamjstewart — DISMISSED** [review]
Python bits look good to me

File: var/spack/repos/builtin/packages/py-amrex/package.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        depends_on("amrex dimensions=3")
+    with when("+mpi"):
+        depends_on("amrex +mpi")
+    with when("+openmp"):
+        depends_on("amrex +openmp")
+    with when("+tiny_profile"):
+        depends_on("amrex +tiny_profile")
+    with when("+cuda"):
+        depends_on("amrex +cuda")
+        # todo: how to forward cuda_arch?
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/py-amrex/package.py:60]: @adamjstewart do you know how I can forward the `cuda_arch` variant here from `CudaPackage`, without doing a long switch statement of all possible values?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-amrex/package.py:60]: I don't do variant forwarding in my packages, too much of a hassle. I wish someone would implement this at the concretizer level (@alalazo?). I just tell people to set things like cuda_arch in `packages:all:variants`.

File: var/spack/repos/builtin/packages/py-amrex/package.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from spack.package import *
+
+
+class PyAmrex(PythonPackage, CudaPackage, ROCmPackage):
+    """AMReX Python Bindings with pybind11"""
+
+    homepage = "https://amrex-codes.github.io/amrex/"
+    git = "https://github.com/AMReX-Codes/pyamrex.git"
+
+    maintainers = ["ax3l", "RTSandberg", "sayerhs", "WeiqunZhang"]
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/py-amrex/package.py:15]: @RTSandberg @sayerhs @WeiqunZhang please let me know if you are ok to be listed as `py-amrex` Spack package co-maintainers. Background: I need someone to approve my PRs to Spack when I update the package (and like to keep the bus-factor >1) :)

Comment (ax3l) [reply var/spack/repos/builtin/packages/py-amrex/package.py:15]: @etpalmer63 likewise, please let me know if you like to be co-listed :)

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-amrex/package.py (hunk 2 above)
Comment (etpalmer63) [reply var/spack/repos/builtin/packages/py-amrex/package.py:15]: @ax3l, I recently transferred out of AMReX :cry: , so its probably better to leave me off. Thanks for asking!

Comment (WeiqunZhang) [reply var/spack/repos/builtin/packages/py-amrex/package.py:15]: Sure. Happy to be a co-maintainer.

**Review by WeiqunZhang — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-amrex/package.py (hunk 2 above)
Comment (RTSandberg) [reply var/spack/repos/builtin/packages/py-amrex/package.py:15]: Ok, I can be a co-maintainer

**Pull request merged**